### PR TITLE
SWC-6105 - Get annotations for specified version in AnnotationsTable.tsx

### PR DESCRIPTION
**Pull request by nickgros**
SWC-6105
- Get annotations for specified version in AnnotationsTable.tsx
- Simplify entity query keys

### CODE REVIEW

File: packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/SchemaDrivenAnnotationEditor.tsx [hunk 1]
```diff
@@ -150,6 +150,7 @@ export function SchemaDrivenAnnotationEditor(
 
   const { data } = useGetJson(
     entityId!,
+    undefined,
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/SchemaDrivenAnnotationEditor.tsx:153]: For the annotation editor, always get the annotations for the latest version

File: packages/synapse-react-client/src/components/entity/metadata/AnnotationsTable.tsx [hunk 2]
```diff
@@ -35,19 +35,18 @@ function getDisplayedAnnotation(value: string | number | boolean): string {
...
    */
   const { isInExperimentalMode } = useSynapseContext()
 
-  // TODO: Support versioned annotations, see PLFM-7290
   const {
     data: entityData,
     isLoading,
     refetch: refetchEntityData,
-  } = useGetJson(entityId, true)
+  } = useGetJson(entityId, versionNumber, true)
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/entity/metadata/AnnotationsTable.tsx:49]: For the AnnotationsTable (read-only), use the prop to get the current version of the annotations

File: packages/synapse-react-client/src/synapse-queries/KeyFactory.ts [hunk 3]
```diff
@@ -44,17 +44,18 @@ const entityQueryKeyObjects = {
...
+  entity: (id: string, versionNumber?: string | number) => {
+    const key: Record<string, unknown> = {
+      ...entityQueryKeyObjects.all,
+      id: id,
+    }
+    if (versionNumber) {
+      key['version'] = versionNumber
+    }
+    return key
+  },
```
Comment (nickgros) [inline packages/synapse-react-client/src/synapse-queries/KeyFactory.ts:56]: Change the query key so our query keys look like 
```
[ { objectType: 'entity', id: 'syn123' } ]
[ { objectType: 'entity', id: 'syn123', version: 2 } ]

[ { objectType: 'entity', id: 'syn123' }, 'activity' ]
[ { objectType: 'entity', id: 'syn123', version: 2 }, 'activity' ]
```

I verified that the invalidations will work as expected. For example, calling `queryClient.invalidateQueries` with the query key  `[ { objectType: 'entity', id: 'syn123' } ]` will invalidate all queries related to syn123, including those that include a version number.

File: packages/synapse-react-client/src/synapse-queries/KeyFactory.ts [hunk 4]
```diff
@@ -44,17 +44,18 @@ const entityQueryKeyObjects = {
...
   // List of Entity Versions
-  versions: (id: string) => ({
-    ...entityQueryKeyObjects.entity(id),
-    scope: 'versions',
```
Comment (nickgros) [inline packages/synapse-react-client/src/synapse-queries/KeyFactory.ts:56]: I removed the `scope` property from the key and moved the value to be the next array item. This is effectively the same, I think it's a bit easier to reason with when the entity id/version are kept in a separate object from this label

File: packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts [hunk 5]
```diff
@@ -301,7 +304,7 @@ export function useUpdateViaJson(
       await invalidateAllQueriesForEntity(queryClient, keyFactory, entityId)
       queryClient.setQueryData(
         // This annotation data will never include derived annotations, which are calculated by the backend asynchronously
-        keyFactory.getEntityJsonQueryKey(entityId, false),
+        keyFactory.getEntityJsonQueryKey(entityId, undefined, false),
```
Comment (hallieswan) [inline packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts:307]: Just to double check, but based on the updates to the query keys, this should invalidate all queries for the entity, including those with the version number?

**Review by hallieswan — APPROVED** [review]
Looks good to me!

File: packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts (hunk 5 above)
Comment (nickgros) [reply packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts:307]: The invalidation will invalidate all queries for the entity. This `setQueryData` will only apply to the "versionless"/"current" annotations data

**Pull request merged**